contrib/babelfishpg_tsql/src/pl_exec-2.c

```diff
@@ -3687,6 +3688,102 @@ get_insert_bulk_kilobytes_per_batch()
 	return insert_bulk_kilobytes_per_batch;
 }
 
+static int
+exec_stmt_grantschema(PLtsql_execstate *estate, PLtsql_stmt_grantschema *stmt)
+{
+	char		*dbname = get_cur_db_name();
+	char		*login = GetUserNameFromId(GetSessionUserId(), false);
+	bool		login_is_db_owner;
+	char		*schema_name;
+	ListCell	*lc;
+	ListCell	*lc1;
+	Oid		schemaOid;
+	char		*user = GetUserNameFromId(GetUserId(), false);
+	const char	*db_owner = get_owner_of_db(dbname);
+
+	login_is_db_owner = 0 == strncmp(login, db_owner, NAMEDATALEN);
+	schema_name = get_physical_schema_name(dbname, stmt->schema_name);
+
+	if(schema_name)
+	{
+		schemaOid = LookupExplicitNamespace(schema_name, true);
+	}
+	else
+	{
+		ereport(ERROR,
+					(errcode(ERRCODE_UNDEFINED_SCHEMA),
+					 errmsg("An object or column name is missing or empty. For SELECT INTO statements, verify each column has a name. For other statements, look for empty alias names. Aliases defined as \"\" or [] are not allowed. Change the alias to a valid name.")));
+	}
+
+	if (!OidIsValid(schemaOid))
+			ereport(ERROR,
+					(errcode(ERRCODE_UNDEFINED_SCHEMA),
+					 errmsg("schema \"%s\" does not exist",
+							stmt->schema_name)));
+
+	foreach(lc1, stmt->privileges)
+	{
+		char	*priv_name = (char *) lfirst(lc1);
```

Comment:
Update this after changing list to integer

Reply:
Why is list of char* a problem here? If we store list of integers, we would need to call `privilege_to_string` here to get the priv_name and generate subcommands.

Reply:
CHeck: https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/2215#discussion_r1455631001